ledger/participant-integration-api/src/main/scala/platform/index/IndexServiceImpl.scala

```diff
@@ -379,16 +400,75 @@ private[index] class IndexServiceImpl(
     }
   }
 
-  private def convertFilter(filter: TransactionFilter): Map[Party, Set[Identifier]] =
-    filter.filtersByParty.map { case (party, filters) =>
-      party -> filters.inclusive.fold(Set.empty[Identifier])(_.templateIds)
-    }
-
   private def concreteOffset(startExclusive: Option[LedgerOffset.Absolute]): Future[Offset] =
     startExclusive
       .map(off => Future.fromTry(ApiOffset.fromString(off.value)))
       .getOrElse(Future.successful(Offset.beforeBegin))
 
   private def toAbsolute(offset: Offset): LedgerOffset.Absolute =
     LedgerOffset.Absolute(offset.toApiString)
+
+  private def withValidatedFilter[T](domainTransactionFilter: domain.TransactionFilter)(
```

Comment:
I assume the reason why the validation doesn't happen in `TransactionFilterValidator` is because you don't have access to `packageMetadataView` there? Otherwise `TransactionFilterValidator` would sound like a better place for doing such up-front checks.

Reply:
I am not entirely sold, that we should pull up service in memory state to the grpc layer (that one is fairly decoupled ATM), also for the better or the worse we are already doing something similar in IndexServiceImpl (look for `Source.failed`).
But good point, it would be cleaner to add to TransctionFilterValidator :+1: